### ISSUE TITLE
Missing $id prefix in $ref

### ISSUE THREAD

**Issue by wizardpisces** [vega/ts-json-schema-generator#1732]
### How to produce

`ts-json-schema-generator -f './tsconfig.json' -p './type.ts'  -o './scheme.json' --id='api'`

./type.ts

```

export type ApiSchema = {
    N: B
}

export type B = string

```

### What to expect?

./schema.json
```

{
  "$id": "api",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "ApiSchema": {
      "type": "object",
      "properties": {
        "N": {
          "$ref": "api#/definitions/B" // with api prefix
        }
      },
      "required": [
        "N"
      ],
      "additionalProperties": false
    },
    "B": {
      "type": "string"
    }
  }
}
```

### What is actually happening?

./schema.json
```json

{
  "$id": "api",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "ApiSchema": {
      "type": "object",
      "properties": {
        "N": {
          "$ref": "#/definitions/B" // missing api prefix
        }
      },
      "required": [
        "N"
      ],
      "additionalProperties": false
    },
    "B": {
      "type": "string"
    }
  }
}
```
Is this by design?

I think it's a serious bug, because it sabotages the combination with ajv validation

_By the way typescript-json-schema  can generate the expected result_

**Comment by domoritz** [comment]
Huh, can you look into why ID is being ignored?

**Comment by wizardpisces** [comment]
Searched source code and get below result:

In ts-json-schema-generator
<img width="1335" alt="image" src="https://github.com/vega/ts-json-schema-generator/assets/3377557/6b30b059-2a9e-44ba-88d9-85b854a223b3">
DefinitionTypeFormatter and ReferenceTypeFormatter hard coded '#/definition' as prefix

In typescript-json-schema
<img width="662" alt="image" src="https://github.com/vega/ts-json-schema-generator/assets/3377557/840ad3e7-f013-4ab9-a4f6-70e7a9c9ae36">
$id will be used as prefix

Since ts-json-schema-generator is an extended version of typescript-json-schema, so I think their behavior should be same?

**Comment by wizardpisces** [comment]
> Huh, can you look into why ID is being ignored?

No idea why it's designed like this, but I searched source code and got below discover

In ts-json-schema-generator, DefinitionTypeFormatter and ReferenceTypeFormatter hard coded '#/definition' as prefix
<img width="1335" alt="image" src="https://github.com/vega/ts-json-schema-generator/assets/3377557/6b30b059-2a9e-44ba-88d9-85b854a223b3">

While in typescript-json-schema, $id will be used as prefix
<img width="662" alt="image" src="https://github.com/vega/ts-json-schema-generator/assets/3377557/840ad3e7-f013-4ab9-a4f6-70e7a9c9ae36">


Since ts-json-schema-generator is an extended version of typescript-json-schema, so I think their basic behavior should be same?

**Comment by domoritz** [comment]
Yeah, looks like we accidentally broke that at some point. Let's fix it.

**Comment by arthurfiorette** [comment]
Would you like to start a PR? I'm happy to review it
